src/Compilers/Core/Portable/Symbols/ISymbol.cs

```diff
@@ -127,7 +127,7 @@ public interface ISymbol : IEquatable<ISymbol?>
 
         /// <summary>
         /// Returns true if this symbol was automatically created by the compiler, and does not have
-        /// an explicit corresponding source code declaration. 
+        /// an explicit corresponding source code declaration. This doesn't include the generated method and class for top-level statements.
```

Comment:
I think the key point is that these are not generated methods. A generated method and class is something like `PrivateImplementationDetails`. The top-level `<Program>$` and `<Main>$` are created because of a specific user action: writing a top-level main. That _means_ create these things. It's not implicitly created.

If we need to add a note, I'd prefer it go down in the remarks below.